src/abstractdataframe/subset.jl

```diff
@@ -0,0 +1,231 @@
+1×5 DataFrame
+│ Row │ id    │ x    │ y    │ z     │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool? │ Int64 │
+├─────┼───────┼──────┼──────┼───────┼───────┤
+│ 1   │ 1     │ 1    │ 1    │ 1     │ 1     │
+
+julia> subset(df, :x, :z)
+ERROR: ArgumentError: each transformation must return a vector whose eltype is subtype of Bool
+
+julia> subset(groupby(df, :y), :v => x -> x .> minimum(x))
+2×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 2     │ 0    │ 1    │ 1       │ 2     │
+│ 2   │ 4     │ 0    │ 0    │ missing │ 12    │
+```
+"""
+@inline function subset(df::AbstractDataFrame, @nospecialize(args...);
+                        skipmissing::Bool=false, view::Bool=false)
+    row_selector = _get_subset_conditions(df, args, skipmissing)
+    return view ? Base.view(df, row_selector, :) : df[row_selector, :]
+end
+
+@inline function subset(gdf::GroupedDataFrame, @nospecialize(args...);
+                        skipmissing::Bool=false, view::Bool=false)
+    row_selector = _get_subset_conditions(gdf, args, skipmissing)
+    df = parent(gdf)
+    return view ? Base.view(df, row_selector, :) : df[row_selector, :]
+end
+
+"""
+    subset!(df::AbstractDataFrame, args...; skipmissing::Bool=false)
+    subset!(gdf::GroupedDataFrame{DataFrame}, args..., skipmissing::Bool=false)
+
+Update data frame `df` or the parent of `gdf` in place to contain only rows for
+which all values produced by transformation(s) `args` for a given row is `true`.
+
+If `skipmissing=true` returing `missing` in `args` is allowed and these rows are
+dropped. If `skipmissing=false` (the default) an error is thrown if `missing` is
+present.
+
+Each argument passed in `args` can be either a single column selector or a
+`source_columns => function` transformation specifier following the rules
+described for [`select`](@ref).
+
+Note that as opposed to [`filter!`](@ref) the `subset!` function works on whole
+columns (or all rows in groups for `GroupedDataFrame`) and by default drops rows
+for which contition is false.
+
+If `GroupedDataFrame` is subsetted then it must include all groups present in the
+`parent` data frame, like in [`select!`](@ref).
+
+See also: [`subset`](@ref), [`filter`](@ref), [`filter!`](@ref)
+
+# Examples
+
+```
+julia> df = DataFrame(id=1:4, x=[true, false, true, false], y=[true, true, false, false])
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
+
+julia> subset!(copy(df), :x, :y => ByRow(!));
+
+julia> df
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
```

Comment:
It's weird to `copy(df)` to illustrate in-place mutation. :-D

Reply:
of course - fixed.